jtuples/src/test/java/org/jtuples/SextupleTest.java

```diff
@@ -0,0 +1,220 @@
+        assertNull(nullSextuple().fourth());
+    }
+    
+    @Test
+    public void testFifth() {
+        assertEquals("5", sextuple.fifth());
+    }
+
+    @Test
+    public void testFifthCanBeNull() {
+        assertNull(nullSextuple().fifth());
+    }
+    
+    @Test
+    public void testSixth() {
+        assertEquals("6", sextuple.sixth());
+    }
+
+    @Test
+    public void testSixthCanBeNull() {
+        assertNull(nullSextuple().sixth());
+    }
+
+    @Test
+    public void testInvert() {
+        expected = new Sextuple("6", "5", "4", "3", "2", "1");
+        
+        result = sextuple.invert();
+
+        assertEquals(expected, result);
+    }
+
+    @Test
+    public void testShiftLeft() {
+        expected = new Sextuple<>("2", "3", "4", "5", "6", "1");
+        
+        result = sextuple.shiftLeft();
+        
+        assertEquals(expected, result);
+    }
+
+    @Test
+    public void testShiftRight() {
+        expected = new Sextuple<>("6", "1", "2", "3", "4", "5");
+        
+        result = sextuple.shiftRight();
+        
+        assertEquals(expected, result);
+    }
+
+    @Test
+    public void testToArray() {
+        Object[] expectedArray = new Object[] { "1", "2", "3", "4", "5", "6" };
+        
+        Object[] resultArray = sextuple.toArray();
+        
+        assertArrayEquals(expectedArray, resultArray);
+    }
+    
+    @Test
+    public void testApplyFirst() {
+        result = sextuple.applyFirst(s -> s + s);
+
+        assertEquals(result.first(), "11");
+    }
+
+    @Test
+    public void testApplySecond() {
+        result = sextuple.applySecond(s -> s + s);
+
+        assertEquals(result.second(), "22");
+    }
+
+    @Test
+    public void testApplyThird() {
+        result = sextuple.applyThird(s -> s + s);
+
+        assertEquals(result.third(), "33");
+    }
+
+    @Test
+    public void testApplyFourth() {
+        result = sextuple.applyFourth(s -> s + s);
+
+        assertEquals(result.fourth(), "44");
+    }
+
+    @Test
+    public void testApplyFifth() {
+        result = sextuple.applyFifth(s -> s + s);
+
+        assertEquals(result.fifth(), "55");
+    }
+
+    @Test
+    public void testApplySixth() {
+        result = sextuple.applySixth(s -> s + s);
+
+        assertEquals(result.sixth(), "66");
+    }
+
+    @Test
+    public void testEqualsIsTrueWhenEqual() {
+        Tuple other = new Sextuple<>("1", "2", "3", "4", "5", "6");
+        
+        assertTrue(sextuple.equals(other));
+    }
+
+    @Test
+    public void testEqualsIsFalseWhenNotEqual() {
+        Tuple other = new Sextuple<>("2", "1", "3", "4", "5", "6");
+        
+        assertFalse(sextuple.equals(other));
+    }
+
+    @Test
+    public void testEqualsIsFalseWhenTypesAreDifferent() {
+        Tuple other = new Sextuple<>("1", "2", "3", "4", "5", 123);
+        
+        assertFalse(sextuple.equals(other));
+    }
+
+    @Test
+    public void testHashCodeIsTheSameForEqualQuintuples() {
```

Comment:
`ForEqualSextuples`, I guess. :wink: 
Maybe `ForEqualTuples` is also a good alternative, to promote copy-paste for future classes.